Add storageType field to HouseTable Entity

ctrezzo opened the pull request:
## Summary

<!--- HINT: Replace #nnn with corresponding Issue number, if you are fixing an existing issue -->

[Issue](https://github.com/linkedin/openhouse/issues/#80) Added a storageType field to the HouseTable Entity to support multiple storage types.

Resolves #80 

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [x] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [x] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

Added storageType field to the HouseTable DTO.

## Testing Done
<!--- Check any relevant boxes with "x" -->

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [x] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

jainlavina reviewed (commented):
We should add a test to check backwards compatibility. Check that if storageType is not populated in the database then it does not break anything.

jainlavina commented on iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/model/HouseTable.java:
```diff
@@ -39,4 +39,6 @@ public class HouseTable {
 
   private long creationTime;
+
+  private String storageType;
```
We don't have a lot of comments here describing the fields. At least this one deserves some explanation with pointer to the list of supported types.

jainlavina replied:
Also, has this column been already added to the schema of all deployed databases?

HotSushi replied:
this column has been added to the deployed database schema already

ctrezzo commented:
Please ignore pull update. Fixing config issues and then will re-update request.

ctrezzo commented:
Hey @HotSushi! I have updated the pull request. Please take a look. Here are the main changes:

1. storageType field is now populated with the HDFS default
2. I added a unit test to validate defaults are set in the HouseTable builder properly
3. I did a manual test with docker containers where I created a table via a spark shell and verified that the user_table_row table had the new row with "hdfs" set as the value for the storage_type column.
4. All unit tests pass.

HotSushi commented on iceberg/openhouse/internalcatalog/src/test/java/com/linkedin/openhouse/internal/catalog/model/HouseTableTest.java:
```diff
@@ -0,0 +1,51 @@
+            continue;
+          }
+
+          if (fieldName.equals("storageType")) {
```
Thanks for this unit test!

HotSushi reviewed (commented):
Its looking good, can we add a screenshot of the request received in HTS service,  maybe from the docker log?

ctrezzo commented:
@HotSushi Attached are the screenshots from the docker log during the manual test. Thanks!

<img width="974" alt="Screenshot 2024-05-03 at 2 03 50 PM" src="https://github.com/linkedin/openhouse/assets/940916/9739a3d5-d257-467a-86f5-ac6d3e8efa69">

<img width="1665" alt="Screenshot 2024-05-03 at 2 03 23 PM" src="https://github.com/linkedin/openhouse/assets/940916/6156e57a-f3ff-4473-b8a8-34838103c906">

<img width="1667" alt="Screenshot 2024-05-03 at 2 00 31 PM" src="https://github.com/linkedin/openhouse/assets/940916/ddf30f26-3109-42bd-93ba-d84a47d2ce9f">

ctrezzo commented:
@HotSushi Attaching screenshot from HTS as well.

<img width="1660" alt="Screenshot 2024-05-06 at 12 46 02 PM" src="https://github.com/linkedin/openhouse/assets/940916/6af93390-16f4-4036-a86e-897df231f1f9">

ctrezzo commented:
Taking a look at the failed test

HotSushi reviewed (approved):
LGTM! thanks for the extensive testing in this PR, unit tests and docker screenshots made it very easy to review
